src/components/Amalgamation/ForeignBusiness.vue

```diff
@@ -0,0 +1,191 @@
+        >
+          <v-btn
+            large
+            color="primary"
+            class="mr-3"
+            @click="addForeignBusiness()"
+          >
+            Add
+          </v-btn>
+          <v-btn
+            large
+            outlined
+            color="primary"
+            @click="emitCancel()"
+          >
+            Cancel
+          </v-btn>
+        </v-col>
+      </v-row>
+    </v-form>
+  </section>
+</template>
+<script lang="ts">
+import { Component, Emit, Vue } from 'vue-property-decorator'
+import NestedSelect from '@/components/common/NestedSelect.vue'
+import { AmalgamatingBusinessIF } from '@/interfaces'
+import { AmlRoles, Location } from '@/enums'
+import { CanJurisdictions, IntlJurisdictions } from '@/list-data'
+
+@Component({
+  components: { NestedSelect }
+})
+export default class ForeignBusiness extends Vue {
+  // Refs
+  $refs!: {
+    foreignBusinessForm: any
+  }
+
+  // Local properties
+  jurisdiction = null
+  legalName = null
+  corpNumber = null
+  isCan = false
+
+  // Validation properties
+  jurisdictionErrorMessage = ''
+
+  get isValid (): boolean {
+    return (this.jurisdiction && this.legalName && this.corpNumber)
+  }
+
+  get rules (): Array<(v) => boolean | string> {
+    return [ v => !!v || 'Required.' ]
+  }
+
+  /** The jursidiction select options */
+  get jurisdictionOptions (): Array<any> {
+    const array = [] as Array<any>
+
+    // add in Canadian jurisdictions (not including BC)
+    array.push({ isHeader: true, group: 0, text: 'Canadian' })
+    CanJurisdictions
+      .filter(jur => jur.value !== Location.BC)
+      .forEach(jur => array.push({
+        group: 0,
+        text: jur.text,
+        value: jur.value,
+        separator: (jur.value === Location.FD)
+      }))
+
+    // add in International jurisdictions (not including CA)
+    array.push({ isHeader: true, group: 1, text: 'International' })
+    IntlJurisdictions
+      .filter(jur => jur.value !== Location.CA)
+      .forEach(jur => array.push({
+        group: 1,
+        text: jur.text,
+        value: jur.value,
+        separator: false
+      }))
+
+    return array
+  }
+
+  /** Called when Jurisdiction menu item is changed. */
+  onJurisdictionChange (jurisdiction: any): void {
+    this.jurisdiction = jurisdiction
+    this.isCan = jurisdiction.group === 0
+    this.jurisdictionErrorMessage = this.jurisdiction ? '' : 'Required.'
+  }
+
+  /** Add the amalgamating foreign business. */
+  addForeignBusiness (): void {
+    // Validate
+    this.validate()
+    if (!this.isValid) return
+
+    // Create the amalgamating foreign business object
+    const business = {
+      type: 'foreign',
+      role: AmlRoles.AMALGAMATING,
+      foreignJurisdiction: {
+        region: this.isCan ? this.jurisdiction.text : '',
+        country: this.isCan ? Location.CA : this.jurisdiction.value
+      },
+      legalName: this.legalName,
+      corpNumber: this.corpNumber
+    } as AmalgamatingBusinessIF
```

Comment:
I'm wondering whether this component should know about this data type/block or whether that should be externalized (eg, a mixin, or emit the data to parent who handles it)?

My thought is to keep complexity as localized as possible.

Reply:
For emitting the data to the parent I think it would look something like this:

```
@Emit('add:foreignBusiness')
private emitForeignBusiness (): any {
  return {
    jurisdiction: this.jurisdiction,
    legalName: this.legalName,
    corpNumber: this.corpNumber
  }
}
```

If you think that's fine I can update the code and move the handling to the parent.

Reply:
Yes please!